src/datasets/packaged_modules/webdataset/webdataset.py

```diff
@@ -0,0 +1,195 @@
+import re
+from typing import List
+
+import numpy as np
+import pyarrow as pa
+from packaging import version
+
+import datasets
+
+
+logger = datasets.utils.logging.get_logger(__name__)
+
+
+class Webdataset(datasets.GeneratorBasedBuilder):
+    DEFAULT_WRITER_BATCH_SIZE = 100
+    IMAGE_EXTENSIONS: List[str]  # definition at the bottom of the script
+    ENABLED_BASIC_HANDLERS: List[str]  # definition at the bottom of the script
+
+    def _basic_handlers(self, key, data):
+        from webdataset.autodecode import decoders
```

Comment:
The format is simple enough to avoid this dependency, no?

Reply:
The dependency on webdataset or on webdataset/autoencode.decoders in particular ?

Reply:
On `webdataset`

Reply:
As a first step I think it's fine and will be easy to maintain. It's also a way to make sure we support webdataset completely (even in the future, as long as they don't move their decoders)

Reply:
I think it's nice to avoid introducing new dependencies if possible.

Reply:
It's optional and pretty lightweight, and as a user I think it's ok to require `webdataset` to read Webdataset data. Do you have concerns regarding breaking changes they may have ? or for user experience ?

Reply:
IMO it makes sense to put some effort into keeping dependencies at minimum if possible. 

An optional dependency still requires a `pip install` call (and restarting a notebook's env), so it's not the best user experience.

Reply:
Ah yes and it's not even installed in colab by default.
Let me see if I can re-use their decoders code (webdataset license is BSD-3-Clause license so it's fine)